llvm/lib/Target/RISCV/RISCVFeatures.td

```diff
@@ -10,115 +10,129 @@
 // RISC-V subtarget features and instruction predicates.
 //===----------------------------------------------------------------------===//
 
+class RISCVExtension<string n, int major, int minor, string f, string v,
+                     string d, list<RISCVExtension> i = []>
+  : SubtargetFeature<n, f, v, d,
+                     !foreach(feature, i, !cast<SubtargetFeature>(feature))> {
+  int MajorVersion = major;
+  int MinorVersion = minor;
+  bit Experimental = false;
+}
+
+class RISCVExperimentalExtension<string n, int major, int minor, string f,
```

Comment:
I have one thought: what about not providing `RISCVExperimentalExtension`, instead, we add a scope:
```
let Experimental = true in {
  def ... : RISCVExtension<...>;
  // ...
}
```
We put all experimental extensions in this scope together, and once an experimental extension is ratified, we will ***move it out*** from experimental scope. This make moving an extension out from expermental concrete.

Reply:
That seems like unnecessary file re-arranging to remove it from experimental since you have to move the Predicates too.

It seems more readable to search for the extension and look for the Experimental than to have to scroll up to see what section it is in.

RISCVExperimentalExtension also automatically adds the "experimental-" to the feature name.

Reply:
Make sense to me.